app/database/queries/breeze_buddy/analytics.py

```
@@ -97,21 +108,25 @@ def build_analytics_where_clause(
         date_from = filters["date_from"]
         if isinstance(date_from, datetime):
-            values.append(date_from)
+            values.append(convert_ist_to_utc(date_from))
         else:
-            values.append(datetime.combine(date_from, datetime.min.time()))
+            values.append(
+                convert_ist_to_utc(datetime.combine(date_from, datetime.min.time()))
+            )
         conditions.append(f"lct.call_initiated_time >= ${len(values) + value_offset}")
 
     if "date_to" in filters and filters["date_to"]:
```

Ruff (0.15.9)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)